### ISSUE TITLE
[BUG] Empty items after Concat

### ISSUE THREAD

**Issue by thenhustey** [Senipah/VBA-DynamicArray#7]
**Describe the bug**
Using a 1D BetterArras for storing Filepaths from folder with recursion. Each (sub)folder returns its own BetterArray, which is "merged" with BetterArray from parent folder using Concat method.

Result 1D BetterArray contains a lot of empty Items.

**To Reproduce**
Consider function:

```vb
Public Function GetFilesArrRecusively(ByVal Folder As Folder) As BetterArray
    Dim File As File
    Dim SubFolder As Folder
    Dim TempArray As BetterArray
    Set TempArray = New BetterArray
    
    ' Loops trought all files in given folder
    For Each File In Folder.Files
        ' Test for hidden and system files
        Dim IsHidden As Boolean
        Dim IsSystem As Boolean
        Dim Attributes As Long
        
        Attributes = File.Attributes
        IsHidden = Attributes And 2
        IsSystem = Attributes And 3
        
        If Not (IsHidden Or IsSystem) Then
            Call TempArray.Push(File.Path)
        End If
    Next File
    ' Now all files in the given folder are in the arr
       
    ' Go trought all subfolders of the given folder
    For Each SubFolder In Folder.SubFolders
        Dim SubFolderArr As BetterArray
        Set SubFolderArr = New BetterArray
        Set SubFolderArr = GetFilesArrRecusively(SubFolder)
        
        Dim TempArrClone As BetterArray
        Set TempArrClone = New BetterArray
        Set TempArrClone = TempArray.Clone ' Save state of TempArray before Concat
        
        ' Appends filepaths from subfolder
        If SubFolderArr.Length > 0 Then
            Call TempArray.Concat(SubFolderArr.Items)
            If TempArray.Includes(Empty) Then
                Debug.Print "EMPTY" '<------ Place breakpoint HERE
            End If
        End If
    Next SubFolder
    Set GetFilesArrRecusively = TempArray
End Function
```

**Expected behavior**
1D BetterArray filled with Filepaths (As String). For each file in given folder (except hidden and system files.
**Screenshots**
This are BetterArrays going to Concat - `Call TempArray.Concat(SubFolderArr.Items)` (remember: `TempArr = TempArray.Clone`)
![image](https://user-images.githubusercontent.com/25990093/103943394-4b642100-5132-11eb-83ea-b0d839474875.png)

This is how TempArray looks like after `Concat` (see `Items(40..63)`)
![image](https://user-images.githubusercontent.com/25990093/103943438-62a30e80-5132-11eb-8a24-be8472249420.png)

**Operating System:**
 - Win 10 Pro

**Host Application:**
 - Excel 2013 (32bit)

**Additional context**
I debugged it step by step from my Concat call (line 36 in my code snippet). And when I stepped on your line 1402 (BetterArray.cls v1.7.3), found that `Me.Items` has 40 Items BUT `This.Items` has 64 items - all extra items are Empty.
![image](https://user-images.githubusercontent.com/25990093/103945224-26bd7880-5135-11eb-8c91-b52813fab0c1.png)
![image](https://user-images.githubusercontent.com/25990093/103945296-49e82800-5135-11eb-8f82-d9e7c067b39a.png)

**Comment by Senipah** [comment]
Hi,

I am struggling to reproduce this issue of having blank lines in the middle of the array. I ran your function on the attached files and got a list of the 250 text files with no empty spaces, which makes me think that FSO is returning blank file names for some files in your environment. Could you try to repro on the attached?:
[nested_test_data.zip](https://github.com/Senipah/VBA-Better-Array/files/5788126/nested_test_data.zip)

>when I stepped on your line 1402 (BetterArray.cls v1.7.3), found that Me.Items has 40 Items BUT This.Items has 64 items - all extra items are Empty.

**Please note**: the Capacity and the Length are not the same. The fact that you see empty items in the internal array is **not** a bug - it is actually a **feature**! Each time the capacity is reached, the internal capacity and size of the internal array doubles (notice how the Capacity is a factor of 2). When you retrieve the stored array it only returns you the items up to the `Length` value. This is detailed in the [About](https://senipah.github.io/VBA-Better-Array/home/about.html) page of the documentation.

If you still think there is a BUG with concat please help me repro by trying the test data I have provided or providing me with some.

Thank you. <3

**Comment by Senipah** [comment]
Here's the Excel workbook I used to test by the way. As I say, for me I got no blank entries. Perhaps you could take a look at that and see if we're doing something differently?

[issue.7.xlsm.zip](https://github.com/Senipah/VBA-Better-Array/files/5788782/issue.7.xlsm.zip)

**Comment by Senipah** [comment]
Hi @thenhustey,

Thanks so much for the bug report. After a fair amount of testing I found the bug - a simple silly error on my part. Concat() was calling `this.Items` instead of `InternalItems`.

This is fixed in the following commit: https://github.com/Senipah/VBA-Better-Array/commit/50e119abef7f92580ba7e746c42f3d7526822099

The change is released in 1.7.4

Please let me know if you still have issues and thanks again. 👍
